lib/types/SSITypesBuilder.ts

```diff
@@ -46,14 +46,56 @@ export class SSITypesBuilder {
   private static mapExternalVerifiableCredentialToInternal(externalCredential: IVerifiableCredential) {
     if (externalCredential.vc && externalCredential.iss) {
       const vc: InternalVerifiableCredential = new InternalVerifiableCredentialJwt();
-      return Object.assign(vc, externalCredential);
+      const result = Object.assign(vc, externalCredential);
+      return SSITypesBuilder.setJWTAdditionalFields(result);
     } else if (externalCredential.credentialSubject && externalCredential.id) {
       const vc: InternalVerifiableCredential = new InternalVerifiableCredentialJsonLD();
       return Object.assign(vc, externalCredential);
     }
     throw 'VerifiableCredential structure is incorrect.';
   }
 
+  private static setJWTAdditionalFields(result: InternalVerifiableCredentialJwt & IJwtCredential & IHasProof) {
+    if (result.exp) {
+      const expDate = result.getBaseCredential().credentialSubject?.expirationDate;
+      if (expDate && expDate !== result.exp) {
+        throw new Error('Inconsistent expiration dates');
```

Comment:
Maybe make the messages a bit more clear. `Inconsistent expiration dates between JWT claim (${result.exp) and VC value (${expDate})`.

Similar for the other error messages.